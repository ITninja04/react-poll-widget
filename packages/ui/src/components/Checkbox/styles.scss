@import '../../colors.scss';

.poll-widget-checkbox {
  display: flex;
  align-items: center;
  cursor: pointer;
  user-select: none;
  
  &::before {
    display: inline-flex;
    content: '';
    width: 1.5em;
    height: 1.5em;
    border: 2px solid $darkblack;
    margin-right: 1em;
    
    border-radius: 0.25em;
  }

  &.checked::before {
    background: $black;
    box-shadow: 0px 0px 0px 2px inset $white;
  }

  &:hover:not(.checked)::before {
    background: $darkwhite;
    box-shadow: 0px 0px 0px 2px inset $white;
    transform: scale(1.1);
  }

  &.radio::before {
    border-radius: 1em;
  }
}