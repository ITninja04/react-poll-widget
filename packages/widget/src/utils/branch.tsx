export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;