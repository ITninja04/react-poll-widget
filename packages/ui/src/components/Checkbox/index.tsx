import React from 'react';
import cx from 'classnames';

import './styles.scss';

const Checkbox: React.FC<{
  type?: 'checkbox' | 'radio',
  label: React.ReactNode,
  checked?: boolean,
  onClick?: React.MouseEventHandler;
}> = ({
  type = 'checkbox',
  label,
  checked,
  onClick
}) => (
      <div
        className={cx(
          'poll-widget-checkbox',
          type,
          { checked }
        )}
        onClick={onClick}
      >
        { label }
      </div>
    );

export default Checkbox;