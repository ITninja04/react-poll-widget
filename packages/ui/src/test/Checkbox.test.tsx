import { createSnapshotTest } from "./utils";
import Checkbox from '../components/Checkbox';

createSnapshotTest(
  Checkbox, 
  {
    checked: true,
    label: 'Test'
  },
  'Checkbox'
  )

createSnapshotTest(
  Checkbox,
  {
    type: 'radio',
    checked: true,
    label: 'Test'
  },
  'Radio checkbox'
)