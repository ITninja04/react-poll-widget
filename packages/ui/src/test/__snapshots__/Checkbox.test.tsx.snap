// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Checkbox should render 1`] = `
<DocumentFragment>
  <div
    class="poll-widget-checkbox checkbox checked"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Radio checkbox should render 1`] = `
<DocumentFragment>
  <div
    class="poll-widget-checkbox radio checked"
  >
    Test
  </div>
</DocumentFragment>
`;
